include/core/non_linear_solver.h

```diff
@@ -57,9 +56,21 @@ class NonLinearSolver
   virtual void
   solve(const bool is_initial_step) = 0;
 
+
+  /**
+   * @brief Return the current newton iteration.
+   *
+   */
+  inline unsigned int
+  get_current_newton_iteration()
```

Comment:
```suggestion
  get_current_newton_iteration() const
```

This is a const function. Let's mark it as such.